crates/ef-testing/src/storage/mod.rs

```diff
@@ -228,3 +231,67 @@ pub fn read_balance(
         .get(&storage_key)
         .unwrap())
 }
+
+/// Check what the implementation class hash should be.
+pub fn implementation_class_hash(
+    account_info: &Account,
+    class_hashes: &ClassHashes,
+) -> FieldElement {
+    if account_info.code.is_empty() && account_info.storage.is_empty() {
```

Comment:
here now we check that account has no storage as well for it to be an EOA

Reply:
so according to my testing on goerli, it's possible to have an CA with no bytecode.

So a CA can have no bytecode and no storage, but it can also have no bytecode and a storage (the initi bytecode returns an empty bytes but set some storage slots).

But an empty CA cannot make any update to its storage nor its nonce, so we probably need to check the post state as well to make sure it's CA or EOA

Reply:
tl;dr is: if it has no code and no storage and nonce = 0, in pre and post than we cannot know if it's an EOA or CA, but it probably doesn't matter

Reply:
I am not sure I get the point of this function, can't you just do just as before? Return `class_hashes.eoa_class_hash` or `class_hashes.contract_account_class_hash` in the `if` on line 85?

Reply:
I wanted to do this at first but decided to make it into a function so I can do a unit test, I do agree it's not idiomatic so open to updating this, what do you think? @ClementWalter @greged93

Reply:
I just find it a little weird because it means you branch twice in your code on `if account_info.code.is_empty() && account_info.storage.is_empty() `: once in the function and once before. This just add boiler plate. You could maybe update the function to return `true` or `false` (so like `is_eoa_contract`) and then branch on this?

Reply:
Gotcha yeah I think that's the way to do it, will update accordingly, thanks @greged93 !

Reply:
I don't really like the function as it is now, the `class_hashes` arguments are transparent to the logic.

According to its logic, its core goal is to determine wether or not an account is a eoa. For this, it needs `account_info` only.

I think that you could just avec written a `fn is_account_eoa` and replace 

```
let proxy_implementation_class_hash = if account_info.code.is_empty() {
```

by 

```
let proxy_implementation_class_hash = if is_account_eoa(account_info) {
```

Reply:
Gotcha that's much cleaner, I've simplified it 🙏 sorry for the roundabout